resources/services/ec2/gen.hcl

```diff
@@ -249,4 +249,44 @@ resource "aws" "ec2" "instance_types" {
       path = "github.com/cloudquery/cq-provider-aws/client.ResolveAWSRegion"
     }
   }
-}
\ No newline at end of file
+}
+
+resource "aws" "ec2" "key_pairs" {
+  path = "github.com/aws/aws-sdk-go-v2/service/ec2/types.KeyPairInfo"
+  ignoreError "IgnoreAccessDenied" {
+    path = "github.com/cloudquery/cq-provider-aws/client.IgnoreAccessDeniedServiceDisabled"
+  }
+  deleteFilter "AccountRegionFilter" {
+    path = "github.com/cloudquery/cq-provider-aws/client.DeleteAccountRegionFilter"
+  }
+  multiplex "AwsAccountRegion" {
+    path   = "github.com/cloudquery/cq-provider-aws/client.ServiceAccountRegionMultiplexer"
+    params = ["ec2"]
+  }
+
+  options {
+    primary_keys = ["key_pair_id"]
+  }
+
+  userDefinedColumn "account_id" {
```

Comment:
does Key Pair have ARN?

Reply:
Then you should add a resolver for it, too.

Basically, you should add (and specify this column to e primary key, see lots of examples in the reources):
```hcl
userDefinedColumn "arn" {
    type              = "string"
    description       = "The Amazon Resource Name (ARN) of the Key Pair"
    generate_resolver = true
}
```
Then you'll need to regenerate the implementation and implement the newly generated resolver.
You can find examples of resolvers utilizing helper functions like [this one](https://github.com/cloudquery/cq-provider-aws/blob/main/client/helpers.go#L290) throughout the codebase.

Reply:
The func signature that cq-gen has a return type of `error`. Should I change the sig to return `schema.ColumnResolver` like the example you linked, or should I keep the generated sig and do like `resources/services/iot/iot_topic_rules.go#L1337` and write into the resource?

Reply:
You should do something like https://github.com/cloudquery/cq-provider-aws/blob/main/resources/services/ec2/images.go#L42-L44

I suppose you can just fill in the proper func body for the generated func.